lib/streamlit/runtime/caching/cached_message_replay.py

```diff
@@ -281,9 +281,8 @@ def save_element_message(
         if not runtime.exists():
             return
         if len(self._cached_message_stack) >= 1:
-
             id_to_save = self.select_dg_to_save(invoked_dg_id, used_dg_id)
-            if isinstance(element_proto, Widget):
+            if isinstance(element_proto, Widget) and element_proto.id:
```

Comment:
Can you add a comment about why checking for the presence of a non-empty ID is now necessary?